stackit/internal/data-sources/argus/job/schema.go

```diff
@@ -0,0 +1,152 @@
+				Type:        types.StringType,
+				Required:    true,
+				Validators: []tfsdk.AttributeValidator{
+					validate.StringWith(
+						instances.ValidateInstanceName,
+						"validate argus instance name",
+					),
+				},
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
+			},
+
+			"project_id": {
+				Description: "Specifies the Project ID the Argus instance belongs to",
+				Type:        types.StringType,
+				Required:    true,
+				Validators: []tfsdk.AttributeValidator{
+					validate.ProjectID(),
+				},
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
+			},
+
+			"argus_instance_id": {
+				Description: "Specifies the Argus Instance ID the job belongs to",
+				Type:        types.StringType,
+				Required:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
+			},
+
+			"metrics_path": {
+				Description: "Specifies the job scraping path. Defaults to `/metrics`",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					modifiers.StringDefault(default_metrics_path),
+				},
+			},
+
+			"scheme": {
+				Description: "Specifies the scheme. Default is `https`.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					modifiers.StringDefault(default_scheme),
+				},
+			},
+
+			"scrape_interval": {
+				Description: "Specifies the scrape interval as duration string. Default is `5m`.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					modifiers.StringDefault(default_scrape_interval),
+				},
+			},
+
+			"scrape_timeout": {
+				Description: "Specifies the scrape timeout as duration string. Default is `2m`.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					modifiers.StringDefault(default_scrape_timeout),
+				},
+			},
+
+			"saml2": {
+				Description: "A saml2 configuration block",
+				Optional:    true,
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"enable_url_parameters": {
+						Description: "Should URL parameters be enabled? Default is `true`",
+						Type:        types.BoolType,
+						Optional:    true,
+						Computed:    true,
+						PlanModifiers: []tfsdk.AttributePlanModifier{
+							modifiers.BoolDefault(default_saml2_enable_url_parameters),
+						},
+					},
+				}),
+			},
+
+			"basic_auth": {
+				Description: "A basic_auth block",
+				Optional:    true,
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"username": {
+						Description: "Specifies basic auth username",
+						Type:        types.StringType,
+						Required:    true,
```

Comment:
attributes shouldn't be marked as `Required` in data source schema unless they are used to identify the resource
This should be changed to `Computed`: `true`